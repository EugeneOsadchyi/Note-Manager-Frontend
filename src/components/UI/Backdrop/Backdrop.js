import React from 'react';
import PropTypes from 'prop-types';

import './Backdrop.css';

const Backdrop = ({ show, clicked }) => (
  show ? <div className="Backdrop" role="presentation" onClick={clicked} /> : null
);

Backdrop.propTypes = {
  show: PropTypes.bool.isRequired,
  clicked: PropTypes.func.isRequired,
};

export default Backdrop;
