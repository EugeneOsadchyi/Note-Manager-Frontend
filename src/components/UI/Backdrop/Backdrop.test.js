import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import Backdrop from './Backdrop';

describe('Backdrop', () => {
  const mockClicked = jest.fn();

  describe('when Backdrop is visible', () => {
    const props = {
      show: true,
      clicked: mockClicked,
    };

    const backdrop = shallow(<Backdrop {...props} />);

    it('renders properly', () => {
      expect(shallowToJson(backdrop)).toMatchSnapshot();
    });

    describe('when backdrop clicked', () => {
      beforeEach(() => backdrop.simulate('click'));

      it('triggers clicked() function', () => {
        expect(mockClicked).toHaveBeenCalled();
      });
    });
  });

  describe('when Backdrop is hidden', () => {
    const props = {
      show: false,
      clicked: mockClicked,
    };

    const backdrop = shallow(<Backdrop {...props} />);

    it('renders properly', () => {
      expect(shallowToJson(backdrop)).toMatchSnapshot();
    });

    it('does not contain Backdrop node', () => {
      expect(backdrop.find('.Backdrop').exists()).toBe(false);
    });
  });
});
