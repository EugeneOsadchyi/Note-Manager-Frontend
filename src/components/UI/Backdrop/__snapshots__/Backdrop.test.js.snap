// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Backdrop when Backdrop is hidden renders properly 1`] = `""`;

exports[`Backdrop when Backdrop is visible renders properly 1`] = `
<div
  className="Backdrop"
  onClick={[Function]}
  role="presentation"
/>
`;
